main/src/cgeo/geocaching/utils/CalculationUtils.java

```diff
@@ -0,0 +1,128 @@
+package cgeo.geocaching.utils;
+
+public final class CalculationUtils {
+    private CalculationUtils() {
+        // utility class
+    }
+
+    /**
+     * This simple evaluation algorithm was created by user 'Boann' and released to the public domain on Stack Overflow:
+     * https://stackoverflow.com/questions/3422673/evaluating-a-math-expression-given-in-string-form
+     *
+     * I added the '%' operator myself.
+     * Note that the 'sqrt', 'cos', 'sin' and 'tan' functions don't work in this implementation as
+     * expected as the individual letters are automatically replaced with their corresponding
+     * expressions as they are interpreted as being individual variables.
+     **/
+    public static double eval(final String str) {
+        return new Object() {
```

Comment:
Ooow, my mind just exploded!

As I've said before, this is actually someone else's algorithm and I don't like poking it too much but I'll do my best to get my head around what you just said.

Someone else is more than welcome to take ownership of this algorithm my the way.  I'm kind of not really wanting to be all that responsible for it.